<launch>
    <arg name="joy_enabled" default="true" />
    <arg name="joy_device" default="js0" />

    <group if="$(arg joy_enabled)">
        <param name="joy_node/deadzone" value="0.2" type="double"/>
        <param name="joy_node/coalesce_interval" value="0.01" type="double"/>
        <param name="joy_node/autorepeat_rate" value="50" type="double"/>
        <param name="joy_node/dev" value="/dev/input/$(arg joy_device)" type="string"/>
        <node name="joy_node" pkg="joy" type="joy_node" output="screen" respawn="false" />
    </group>
    
    <node name="xbox_wrapper" pkg="orthopus_interface" type="xbox_wrapper" output="screen" />
    <node name="cartesian_controller" pkg="orthopus_interface" type="cartesian_controller" output="screen" respawn="false" >
        <param name="alpha_1" type="double" value="10.0" />
        <param name="alpha_2" type="double" value="10.0" />
        <param name="alpha_3" type="double" value="10.0" />
        <param name="alpha_4" type="double" value="10.0" />
        <param name="alpha_5" type="double" value="10.0" />
        <param name="alpha_6" type="double" value="10.0" />
        <param name="alpha_7" type="double" value="10.0" />
        
        <param name="beta_1" type="double" value="1e-1" />
        <param name="beta_2" type="double" value="1e-1" />
        <param name="beta_3" type="double" value="1e-1" />
        <param name="beta_4" type="double" value="1e-1" />
        <param name="beta_5" type="double" value="1e-1" />
        <param name="beta_6" type="double" value="1e-1" />
        
        
      <!--      Test without cartesian position control  
        <param name="gamma_1" type="double" value="5000.0" />
        <param name="gamma_2" type="double" value="200.0" />
        <param name="gamma_3" type="double" value="200.0" />
        <param name="gamma_4" type="double" value="5000.0" />
        <param name="gamma_5" type="double" value="5000.0" />
        <param name="gamma_6" type="double" value="100.0" />   
        -->    
        <param name="gamma_1" type="double" value=".0" />
        <param name="gamma_2" type="double" value=".0" />
        <param name="gamma_3" type="double" value=".0" />
        <param name="gamma_4" type="double" value="1000.0" />
        <param name="gamma_5" type="double" value="1000.0" />
        <param name="gamma_6" type="double" value="1000.0" />
        <param name="gamma_7" type="double" value="1000.0" />
    </node>
</launch>
